src/app/components/Ionic/providers/IonicProvider.tsx

```diff
@@ -0,0 +1,30 @@
+import { FC, PropsWithChildren, useState } from 'react'
+import { useIonicBackButtonListener } from '../hooks/useIonicBackButtonListener'
+import { useIonicAppStateChangeListener } from '../hooks/useIonicAppStateChangeListener'
+import { IonicContext, IonicProviderContext, IonicProviderState, UpdateAvailability } from './IonicContext'
+import { useIonicRequiresUpdate } from '../hooks/useIonicRequiresUpdate'
+
+const ionicProviderInitialState: IonicProviderState = {
+  updateAvailability: UpdateAvailability.NOT_INITIALIZED,
+  error: null,
+}
+
+export const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [state, setState] = useState<IonicProviderState>({ ...ionicProviderInitialState })
+
+  const { checkForUpdateAvailability } = useIonicRequiresUpdate(state, setState)
```

Comment:
Sending changes into setState like this is hard to trace

Reply:
Yes I agree. But making a redux slice just for this seemed like an overkill.

Reply:
ew i didn't mean that :D just move useState into useIonicRequiresUpdate and return state out

Reply:
Right, my bad 😄 

Here is the follow up for it: https://github.com/oasisprotocol/oasis-wallet-web/pull/1969